clients/drcachesim/tests/scheduler_unit_tests.cpp

```diff
@@ -6574,11 +7201,14 @@ run_lockstep_simulation_for_kernel_seq(scheduler_t &scheduler, int num_outputs,
                     assert(prev_tid[i] != tid_from_memref_tid(memref.instr.tid));
                 } else {
                     assert(for_syscall_seq || prev_tid[i] == INVALID_THREAD_ID ||
-                           prev_tid[i] == tid_from_memref_tid(memref.instr.tid));
+                           tid_from_memref_tid(prev_tid[i]) ==
+                               tid_from_memref_tid(memref.instr.tid));
                 }
             }
-            prev_tid[i] = tid_from_memref_tid(memref.instr.tid);
-            prev_in_ord[i] = outputs[i]->get_input_interface()->get_record_ordinal();
+            prev_tid[i] = memref.instr.tid;
+            prev_in_ord[i] = outputs[i]->get_input_interface() == nullptr
```

Comment:
This returning null seems like a new thing: do we need a release note on this? If none of our code is checking for it, it could break other users.

Reply:
Opened https://github.com/DynamoRIO/dynamorio/issues/7662 for the nullptr get_input_interface return value which can happen also without this change when there are idle records.